### ISSUE TITLE
combine_ivar doesn't work as advertised

### ISSUE THREAD

**Issue by weaverba137** [desihub/desispec#52]
The test in `desispec.test.test_util.TestNight.test_combine_ivar()` performs this test:

```python
        ivar = util.combine_ivar(1, 2)
```

`desispec.util.combine_ivar` tries to convert the scalar values to an array, using

```python
    iv1 = np.asarray(ivar1)  #- handle list, tuple, ndarray, and scalar input
    iv2 = np.asarray(ivar2)
```

However, for scalar input, `np.asarray()` returns a numpy *scalar*, which cannot be indexed.  When the combined inverse variance is computed, 

```python
    ivar[ii] = 1.0 / (1.0/iv1[ii] + 1.0/iv2[ii])
```

the indexing of the scalar values throws an exception.

It is very strange that this test *ever* passed.

**Comment by weaverba137** [comment]
This may be due to the numpy version.  In 1.9.x, this actually seems to work. However, in 1.8.x it does not.

**Comment by sbailey** [comment]
Good to support numpy 1.8.x, but it would probably be better to handle this using np.atleast_1d() instead of the original np.asarray(), rather than the explicit numpy version checking in commit d4f7735 .

**Comment by weaverba137** [comment]
Ah, yet another useful function that numpy thought of already!

**Comment by weaverba137** [comment]
This should have been automatically closed by #56.  However, I think you have to reference the issue in the *body* of a PR in order to have a PR automatically close it.
